bindings/nodejs/src/client.rs

```diff
@@ -8,26 +8,22 @@ use iota_sdk_bindings_core::{
     iota_sdk::client::{mqtt::Topic, Client, ClientBuilder},
     listen_mqtt as rust_listen_mqtt, ClientMethod, Response,
 };
-use napi::{bindgen_prelude::External, threadsafe_function::ThreadsafeFunction, Error, Result, Status};
+use napi::{bindgen_prelude::External, threadsafe_function::ThreadsafeFunction, Result};
 use napi_derive::napi;
 use tokio::sync::RwLock;
 
-use crate::NodejsError;
+use crate::{build_js_error, destroy, NodejsError};
 
 pub type ClientMethodHandler = Arc<RwLock<Option<Client>>>;
```

Comment:
I feel like we should be able to remove the option and just drop the arc in destroy 🤷‍♀️

Reply:
I dont think so, we have no idea on the references in the js side. and I also dont know how External<> handles things. lests leave it like this for now, at least its cleaner and aligned with the other code.

Reply:
if js is going to leak memory, it's going to do it regardless of how we impl this. I really think there's no point in doing it this way